### PR TITLE
feat: Add support for release archives

### LINKED ISSUE #647: [FEATURE] Support for release archives
Add support for git release archives of ProServe managed modules.  

This will enable:
- confirmed downloads of released modules
- protocol support over HTTPS rather than cloning repos

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
## Description

- Add support for downloading modules from archives (prefixed with `archive::`)
- Supports downloading from private archives using secret credentials
- Support downloading from S3 using signed HTTPS requests

## Issue

Closes #647

---

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: seedfarmer/commands/_deployment_commands.py [hunk 1]
```diff
@@ -56,21 +57,38 @@
...
 
-def _process_module_path(module: ModuleManifest) -> None:
+def _process_git_module_path(module: ModuleManifest) -> None:
     working_dir, module_directory, commit_hash = sf_git.clone_module_repo(module.path)
     module.set_local_path(os.path.join(working_dir, module_directory))
     module.commit_hash = commit_hash if commit_hash else None
 
 
-def _process_data_files(data_files: List[DataFile], module_name: str, group_name: str) -> None:
+def _process_archive_path(module: ModuleManifest, secret_name: Optional[str] = None) -> None:
```
Comment (dgraeber) [inline seedfarmer/commands/_deployment_commands.py:66]: why was the data path removed?  Was this tested with a module using a datapath...such as:

```
name: mwaa
path: modules/orchestration/mwaa
dataFiles:
  - filePath: archive::https://github.com/awslabs/idf-modules.zip?module=data/mwaa/requirements/requirements-emr-serverless.txt
```
Just want to confirm that this was addressed

**Review by dgraeber — DISMISSED** [review]
Looks good but I don't see where you defined in the docs how to create the secret...
ref: https://seed-farmer.readthedocs.io/en/latest/manifests.html#mirror-secrets

File: seedfarmer/commands/_deployment_commands.py (hunk 1 above)
Comment (LeonLuttenberger) [reply seedfarmer/commands/_deployment_commands.py:66]: I tested it with a datapath, it worked. It seems like git diff is just displaying this change in a strange way. I didn't replace `_process_data_files` with `_process_archive_path`. The `_process_data_files` function is still there, it's just after the definition of `_process_archive_path`,

**Review by malachi-constant — DISMISSED** [review]

**Review by malachi-constant — DISMISSED** [review]

**Comment by dgraeber** [comment]
This still needs to be tested with a basic auth repo, correct?

**Comment by LeonLuttenberger** [comment]
> This still needs to be tested with a basic auth repo, correct?

Yes

**Review by dgraeber — APPROVED** [review]

**Review by malachi-constant — APPROVED** [review]

**Pull request merged**